Sources/Bindings/SFTP.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  SFTP.swift
+//  Bindings
+//
+//  Created by Vladislav Alexeev on 6/20/18.
+//
+
+import Foundation
+import CSSH
+
+public class SFTP {
+    private let sftpSession: OpaquePointer
+    
+    public struct Permissions : OptionSet {
+        public let rawValue: UInt
+        public init(rawValue: UInt) {
+            self.rawValue = rawValue
+        }
+        
+        public static let read = Permissions(rawValue: 1)
+        public static let write = Permissions(rawValue: 2)
+        public static let execute = Permissions(rawValue: 3)
+    }
+    
+    public struct FilePermissions {
+        public var owner: Permissions
+        public var group: Permissions
+        public var others: Permissions
+
+        public init(owner: Permissions, group: Permissions, others: Permissions) {
+            self.owner = owner
+            self.group = group
+            self.others = others
+        }
+        
+        public static func fromPosixPermissions(_ value: CShort) -> FilePermissions {
+            var permissions = FilePermissions(owner: [], group: [], others: [])
+            if (value & CShort(S_IRUSR) == CShort(S_IRUSR)) { permissions.owner = permissions.owner.insert(.read).memberAfterInsert }
```

Comment:
Looking at the docs of `.insert`, it says it returns:

```
(true, newMember)` if `newMember` was not contained in
    ///   `self`. Otherwise, returns `(false, oldMember)`, where `oldMember` is
    ///   the member of the set equal to `newMember`."
```

which makes it seem to me as though `.insert(.read)` will always return `.read`, overwriting `.owner`. Should these lines just be:

```swift
if (value & CShort(S_IRUSR) == CShort(S_IRUSR)) { permissions.owner.insert(.read) }
```

? I haven't worked much with option sets so apologies if I misunderstood this line

Reply:
You're absolutely right. Updated my PR.